### PR TITLE
feat: add pay-to-relay

### PR DESCRIPTION
**Pull request by Cameri**
## Screenshots (if appropriate):
image

image

image

## Types of changes

- [ ] Non-functional change (docs, style, minor refactor)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my code changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .nostr.local/settings.json [hunk 1]
```diff
@@ -1,19 +1,70 @@
 {
   "info": {
-    "relay_url": "wss://nostream.localtest.me",
-    "name": "nostream.localtest.me",
-    "description": "A nostr relay written in TypeScript.",
-    "pubkey": "replace-with-your-pubkey",
+    "relay_url": "wss://nostr-relay-dev.wlvs.space",
```
Comment (Cameri) [inline .nostr.local/settings.json:3]: will revert once I'm done testing
this file should probably not be committed anyway?

**Review by kevinsmith — COMMENTED** [review]
Wanted to send you the comments so far, but I'm gonna have to finish this up tomorrow.

File: .nostr.local/settings.json [hunk 2]
```diff
@@ -1,19 +1,70 @@
...
+          }
+        }
+      ]
+    }
+  },
+  "paymentProcessors": {
+    "zebedee": {
+      "baseURL": "https://api.zebedee.io/",
+      "callbackBaseURL": "https://nostr-relay-dev.wlvs.space/callbacks/zebedee"
+    }
```
[the remark this answers is not in the document]
Comment (Cameri) [reply .nostr.local/settings.json:42]: We may have to tweak a few things to make it work with CLN, LND, BTCPayServer, etc. for sure.
I have not looked deeply at how each payment processor works but I tried to make the interfaces in such a way that it's easy to swap between them.

This `paymentsProcessors` object should allow for adding specific settings for each one and is not shared amongst each other (i.e. zebedee code will only access the zebedee setting here)

File: scripts/stop [hunk 3]
```diff
@@ -2,8 +2,10 @@
 PROJECT_ROOT="$(dirname $(readlink -f "${BASH_SOURCE[0]}"))/.."
 DOCKER_COMPOSE_FILE="${PROJECT_ROOT}/docker-compose.yml"
 DOCKER_COMPOSE_TOR_FILE="${PROJECT_ROOT}/docker-compose.tor.yml"
+DOCKER_COMPOSE_LOCAL_FILE="${PROJECT_ROOT}/docker-compose.local.yml"
```
[the remark this answers is not in the document]
Comment (Cameri) [reply scripts/stop:5]: Hehe this one bit me a few times.

File: docker-compose.yml [hunk 4]
```diff
@@ -19,6 +20,8 @@ services:
       # Enable DEBUG for troubleshooting. Examples:
       # DEBUG: "worker:*"
       # DEBUG: "knex:query"
+    env_file:
+      - test.env
```
[the remark this answers is not in the document]
Comment (Cameri) [reply docker-compose.yml:24]: Yeah, `.env` files are ignored in the .gitignore file. I have ZEBEDEE API keys and secrets in mine and don't want those ending up in GitHub.
These two lines will be removed before merging.

**Comment by bleebl33** [comment]
Afaict, relays that require payments always appear to be offline on monitors like nostr.watch. Is it possible to instead reserve nodejs workers for registered client pubkeys? Suppose you have a dedicated server with 16+ cpu cores, and have 12 workers answer public requests while reserving 4 workers to be available anytime a registered client makes a request? That way anyone with a public relay can provide higher reliability to clients that have registered by making a lightning payment.

ie. in settings.json
```json
  "workers": {
    "public": 12,
    "private": 4
  },
```

**Pull request merged**